### DISCUSSION TITLE
Change text bar without affect tabs

### DISCUSSION THREAD

**Discussion by meroxdotdev** [squidfunk/mkdocs-material#7378] [Q&A] [answered]
<img width="952" alt="Screenshot 2024-07-19 at 23 12 30" src="https://github.com/user-attachments/assets/ca8de176-6bb6-4507-99c9-64e12de70e9c">

Hello, please tell me how can I change the text which is after the logo ( apple avatar ) MEROX.DEV to something else, and also keep the MEROX.DEV in every page tab opened ( like Home - MEROX.DEV )? Thank you!

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @mer0x,
the value in the browser tab comes from the `title` tag in the `head` part of the page, like in all other websites:
https://github.com/squidfunk/mkdocs-material/blob/6b13c560f5bf76889b1c603a5b83941892279c76/src/templates/base.html#L89-L98
The value next to the logo in the header can change based on the scroll and heading:
https://github.com/squidfunk/mkdocs-material/blob/6b13c560f5bf76889b1c603a5b83941892279c76/src/templates/partials/header.html#L55-L73

In both cases the `MEROX.DEV` comes from the `site_name` value in your `mkdocs.yml`, so you need to override one case.
Since you want to keep the value in the tab name, I would recommend overriding the `htmltitle` block.
Just copy-paste the contents into your `main.html` block override and replace the `{{ config.site_name }}` with your desired text value.
- https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks

**Reply by meroxdotdev** [reply]
Hello, thank you for your help, so I added this:

```
{% block header %}
<header class="md-header">
  <div class="md-header__inner">
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">My Custom Text</span>
        </div>
      </div>
    </div>
    {% if config.repo_url %}
    <a href="{{ config.repo_url }}" class="md-header-nav__button md-logo" title="GitHub">
      <i class="icon">link</i>
    </a>
    {% endif %}
  </div>
</header>
{% endblock %}
```

in main.html, now it gets better, but the logo dissapear and the text position is not the same as before ( screenshot attached )
<img width="889" alt="Screenshot 2024-07-20 at 09 13 16" src="https://github.com/user-attachments/assets/77ebe44c-e178-435f-b578-39c8822af6e4">

**Reply by meroxdotdev** [reply]
<details>
<summary>Screenshot</summary>

![image](https://github.com/user-attachments/assets/ef19d4d3-683c-4249-b14b-0885d1ba1ac7)

</details>

Also from mobile, the burger menu is not displayed and in right corner appear "link" which is going to github repo

**Reply by kamilkrzyskow** [reply]
Why did you add a `header` block when I suggested the `htmltitle` block which should be simpler to maintain?
Also your approach is incorrect, as it removes too much. The `header` block in the `base.html` includes the whole `header.html` partial.
https://github.com/squidfunk/mkdocs-material/blob/6b13c560f5bf76889b1c603a5b83941892279c76/src/templates/base.html#L270-L273

So if you want to modify the `header`, then create an exact copy of the file first and put it under the matching path in your custom_dir and only then start editing the custom partial file.
- https://github.com/squidfunk/mkdocs-material/blob/master/material/templates/partials/header.html

**Reply by meroxdotdev** [reply]
Hello, thank you for clarification, this is how I configured now:

https://github.com/mer0x/merox.dev/blob/master/assets/main.html

But doesnt look like how I want

<details>

![image](https://github.com/user-attachments/assets/3ca46a76-68b9-46d0-b743-8531454270b9)

</details>

This implementation just added that black bar, maybe I still implemented wrong, can you check please? Thanks!

**Reply by kamilkrzyskow** [reply]
Focus on the example in the OP first, mobile comes later (aka I think should not require any attention)
The `htmltitle` block is located in the `<head>` tag, there should not be any divs there, also by design this block should render the one single `<title>` tag, not some elaborate structure.
- https://www.w3schools.com/TAGS/tag_title.asp

I think I elaborated before enough:
> Just copy-paste the contents into your main.html block override and replace the {{ config.site_name }} with your desired text value.

1. Copy this into your custom block, and replace the config value with a static string `MEROX.DEV`
    To meet the criteria:
    > and also keep the MEROX.DEV in every page tab opened ( like Home - MEROX.DEV 

    https://github.com/squidfunk/mkdocs-material/blob/6b13c560f5bf76889b1c603a5b83941892279c76/src/templates/base.html#L91-L97

    So the custom `htmltitle` will use a custom static text in the template, in-place of the loaded config value `{{ config.site_name }}`.

2. And change the `site_name` back to the alternative to use it at the top of the page.
`site_name: root@merox.dev:~#`

It's not the first time you're asking for override tips on the forums, at this point I feel like you should have HTML/CSS 101 under your tool belt.

**Reply by kamilkrzyskow** [reply]
Also I see you're doing a lot of commits, I highly recommend installing Python + mkdocs on your local machine and test things in your browser. Currently it looks like you're doing changes blindly and then waiting for the CI to push the changes onto the server to see the changes, so instead of 3 seconds you wait 80s+

**Reply by meroxdotdev** [reply]
Thank you for your support, I managed it! Working as expected now.

Yes, for sure I need to create an test env ( I was a little bit lazy to do this, but it will be the next step )

All the best 🍺
